### DISCUSSION TITLE
How to get the number of connections to the OPC server

### DISCUSSION THREAD

**Discussion by evanZhangYiFeng** [eclipse-milo/milo#1010] [General]
Hello dear author! Recently I use your milo frame in my SCADA project . In my project I use connection pool to cache my opcUaClient connections,  but in some cases I afraid of the number of connection pools is insufficient, so I want to know the number of connections to the current server so that I can dynamic increase the number of the connection pool and this number is less than the OPC server's maximum limit. Finally looking forward to your reply.

**Comment by kevinherron** [comment]
I think you’ll just have to keep track of how many connections you’ve made. 

If a server supports diagnostics (and you enable them or they are already enabled) you can probably figure out the total number of sessions open to that server, including other clients.

That said… it’s very unlikely you even need a connection pool at all. I’ve never seen that be necessary and most client connections are persistent, not constantly being connected and disconnected.
